source/agora/test/Base.d

```diff
@@ -175,6 +177,9 @@ public interface TestAPI : API
 
     ///
     public abstract void metaAddPeer(string peer);
+
+    ///
+    public abstract void addBlockToLedger (Block block);
```

Comment:
Why do you need this ? I don't see an override anywhere ?

Reply:
Think about it. How do you get access to the Ledger from the unittest.

Reply:
You never access the internals of a node at the moment.

Reply:
Well the first test is:

Instantiate the nodes. Don't start networking yet. One node (node A) needs to have 100 blocks, the 3 are empty.

Then, make them connect to each other. The 3 other nodes will retrieve blocks from the 1st node on boot.

But to make this work, I need to add those blocks to node A before it connects to the 3 other nodes.

Reply:
To be more specific here. I can't just use the `sendTransaction` API here because this will trigger the gossip protocol to send the transaction over the whole network, and then each node will create their own block. So by the time it tries to retrieve the blocks from the other nodes, it will already have all the blocks.

Delaying the node might not help. I would need some way to temporarily disable the gossip protocol, but then this becomes very messy to have all these special cases for unittesting.

The more features the node has, the more complicated it is to test a specific feature in the node without triggering an avalance of other features.

Reply:
I will be able to replace this code once LocalRest has the ability to filter requests. Then I can do something like:

- Use the sendTransaction API to one node
- Make the other nodes ignore transaction gossiping
- The other nodes will then catch-up using the periodic `getBlocksFrom` call.

This is way nicer than messing with Node's internals.